src/Contacts/TSThread.m

```diff
@@ -248,9 +248,11 @@ - (NSString *)lastMessageLabel {
 - (void)updateWithLastMessage:(TSInteraction *)lastMessage transaction:(YapDatabaseReadWriteTransaction *)transaction {
     NSDate *lastMessageDate = lastMessage.date;
 
-    if ([lastMessage isKindOfClass:[TSIncomingMessage class]]) {
```

Comment:
Yes, I believe this logic is used to update the "last message date" for threads, which determines their ordering in the conversations view.  We've always updated that ordering for any interaction, but we were only using accurate "received" timestamps for incoming messages. Going forward we'll use "received" timestamps for all interactions. That should make the ordering of conversations more consistent.